url8417b1b213_Ratchette_Rummikub_tgz-pJ8-rummikub_RummikubJ8/src/rummikub/HintClient.java

```diff
@@ -0,0 +1,191 @@
+package rummikub;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class HintClient extends Thread{
+	private final int playerIndex;
+	
+	private Scanner keyboard;
+	
+	private GameInfo game;
+	private Hand hand;
+	private boolean initialMeld;
+	private int round;
+
+	// **********************************************************
+	// 					Constructors
+	// **********************************************************
+	
+	public HintClient(boolean initialMeldMade){
+		game = null;
+		hand = null;
+		initialMeld = initialMeldMade;
+		round = 0;
+		
+		// TODO Change to something more meaningful?
+		playerIndex = GameInfo.PLAYER1;
+		keyboard = new Scanner(System.in);
+	}
+	
+	// **********************************************************
+    // 				Main + User input validation
+    // **********************************************************
+	
+	public static void main(String[] args){
+		boolean initial = false;
+		
+		if(args[0].equalsIgnoreCase("true"))
+			initial = true;
+		else if(args[0].equalsIgnoreCase("false"))
+			initial = false;
+		else{
+			System.out.println("Invalid usage of this program.Proper syntax is:");
+			System.out.println("\nProper syntax of this program is ");
+			System.out.println("\tHintClient.java <initial meld played>");
+			System.out.println("\tinitial meld played defaults to false");
+			System.exit(1);
+		}
+		
+		(new HintClient(initial)).run();
+	}
+	
+	// **********************************************************
+    // 						Starting a Game
+    // **********************************************************
+	
+	/**
+	 * Initialize a new game
+	 * @throws Exception
+	 */
+	private void startGame(){ 
+		String userInput;
+		
+		game = new GameInfo(2);
+		
+		while(hand == null){
+			System.out.print("Please enter your initial hand: ");
+			userInput = keyboard.nextLine();
+			
+			try {
+				hand = new Hand(userInput);
+				
+			} catch (Exception e) {
+				hand = null;
+				System.out.println("\t[ INVALID HAND ]: " + userInput);
+				System.out.println("");
+			}
+		}
+		System.out.println("You entered: " + hand.toString());
+	}
+	
+	// **********************************************************
+    // 						Playing a game
+    // **********************************************************
+	
+	public void run(){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 13. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.